detox/src/devices/drivers/android/tools/EmulatorHandle.js

```diff
@@ -5,25 +5,23 @@ class EmulatorHandle extends DeviceHandle {
   constructor(deviceString) {
     super(deviceString);
 
+    this.telnet = new EmulatorTelnet();
     this.port = this.adbName.split('-')[1];
   }
 
-  queryName() {
+  async queryName() {
     if (!this._name) {
-      this._name = this._queryNameViaTelnet();
+      this._name = await this._queryNameViaTelnet();
```

Comment:
So was this a bug, or not? (not related to #2169).

Reply:
@d4vidi , no, don't do it!!! 😱

Reply:
I am caching a promise inside `this._name` and making sure to return always the same instance of a Promise. Please don't touch this code.

Reply:
np, I'll revert - but with the subtle change of renaming `_name` to `_namePromise`.
Is it really that important to cache `name` as a promise and not as the resulted value?